table/tables/tables.go

```diff
@@ -72,12 +72,24 @@ func TableFromMeta(alloc autoid.Allocator, tblInfo *model.TableInfo) (table.Tabl
 		return nil, table.ErrTableStateCantNone.Gen("table %s can't be in none state", tblInfo.Name)
 	}
 
-	columns := make([]*table.Column, 0, len(tblInfo.Columns))
-	for _, colInfo := range tblInfo.Columns {
+	colsLen := len(tblInfo.Columns)
+	columns := make([]*table.Column, 0, colsLen)
+	for i, colInfo := range tblInfo.Columns {
 		if colInfo.State == model.StateNone {
 			return nil, table.ErrColumnStateCantNone.Gen("column %s can't be in none state", colInfo.Name)
 		}
 
+		// Add information about the column's offset.
```

Comment:
I am confused by this comment. Per my understanding, it is first checking whether column's offset is correct or not. If not, print error info. This comment may mislead readers.

Reply:
I don't understand you comments.
Only when it meets the condition of `i == colsLen-1 && colInfo.Offset >= colsLen` it's an error . Then it will print the log with error level.

Reply:
You are absolutely right. I was saying about the comments. It is not "add information about the column's offset", it is check the offset is right or not. If not, print error info. If it is correct, just log normally.